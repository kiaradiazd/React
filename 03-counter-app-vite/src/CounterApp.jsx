import React from 'react';
import { useState } from 'react';
import PropTypes from 'prop-types';

export default function CounterApp({ value }) {

    const [ counter, setCounter ] = useState( value );

    const handleAdd = () => {
        setCounter( counter + 1 );
        //setCounter( (c) => c + 1 )
    }

    const handlead = () => setCounter( counter - 1 );
    

    const handle = () => setCounter( value);

  return (
    <>
    <h1>CounterApp</h1>
    <h2>{ counter }</h2>
    <button onClick={ handleAdd }> +1 </button>
    <button onClick={ handlead }> -1 </button>
    <button aria-label="btn-reset" onClick={ handle }> reset </button>
    </>
  )
}

CounterApp.prototype = {
    value : PropTypes.number.isRequired
}