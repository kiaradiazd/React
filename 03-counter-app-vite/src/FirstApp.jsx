import PropTypes from 'prop-types';

export const FirstApp = ({ title, sudTitle, name }) => {

    return (
        <>
        <div data-testid="test-title">{ title }</div>

        <p>{ sudTitle }</p>
        <p>{ name }</p>
        </>
    )
};

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    sudTitle: PropTypes.string,
};

FirstApp.defaultProps = {
    name: 'Kiara Diaz',
    sudTitle: 'No hay sudtitulo',
    title: 'No hay titulo',
};