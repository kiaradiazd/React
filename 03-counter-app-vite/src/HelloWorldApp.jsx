

export function App() {
}

import React from 'react'

const HelloWorldApp = () => {
    return (
        <div>
        <h1> Hola mundo!!!</h1>
      </div>
    );
  }
  
  export default HelloWorldApp;
  