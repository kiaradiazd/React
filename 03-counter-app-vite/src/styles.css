html, body {
    background-color: bisque;
    color: black;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    padding: 15px;
}