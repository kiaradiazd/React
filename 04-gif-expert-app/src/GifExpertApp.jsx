import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = () => {

    const [ categories, setCategories] = useState( [ 'One Punch' ] );

    const onAddCategory = ( newCategory ) => {

        //valide que no sea igual a los valores que ya estan
        if( categories.includes( newCategory )) return;
        //como aÃ±adir valor a un array y si quieres agregar al inicio poner valorant antes
        setCategories([ newCategory, ...categories ]);
        //otra manera
        //setCategories( cat => [...cat, 'Valorant' ] );
    };


  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory onNewCategory={ onAddCategory }/>
      <ol>
        {
            categories.map( ( category ) => (
                    <GifGrid key={ category } category={ category }/>
            ))
        }
      </ol>
    </>
  )
}

export default GifExpertApp;
