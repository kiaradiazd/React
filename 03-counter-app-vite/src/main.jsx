import React from 'react';
import ReactDOM from 'react-dom/client';
import CounterApp from './CounterApp';
import { FirstApp } from './FirstApp';
import HelloWorldApp from './HelloWorldApp'
import './styles.css'
CounterApp

ReactDOM.createRoot(document.getElementById('root') ).render(
  <React.StrictMode>
   {/*<HelloWorldApp /> */}
    <FirstApp title="hojojojo" />
   {/**  <CounterApp value={0}/>*/}
  </React.StrictMode>
);