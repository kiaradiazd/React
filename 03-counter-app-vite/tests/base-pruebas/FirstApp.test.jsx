import { render, renderHook } from '@testing-library/react';
import { FirstApp } from '../../src/FirstApp';



describe('Pruebas en <FirstApp />', () => {
    /*
    test('debe de hacer mach con el snapshot', () => {
        
        const title = 'Hola soy Vgeta';
        const { container } = render(<FirstApp title={ title } />)
        
        expect( container ).toMatchSnapshot();

    });
    */
    
    test('debe de mosntrar el titulo en un h1', () => {
        
        const title = 'Hola, soy Goku';
        const { container, getByText  } = render( <FirstApp title={ title } />);
        
        expect( getByText(title) ).toBeTruthy();
        
        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain( title )
        
    })
});