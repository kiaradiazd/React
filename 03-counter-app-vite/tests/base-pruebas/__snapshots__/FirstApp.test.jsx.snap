// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer mach con el snapshot 1`] = `
<div>
  <h1>
    Hola soy Vgeta
  </h1>
  <p>
    No hay sudtitulo
  </p>
  <p>
    Kiara Diaz
  </p>
</div>
`;
